k9mail/src/main/java/com/fsck/k9/activity/FolderList.java

```diff
@@ -1166,8 +1166,8 @@ protected FilterResults performFiltering(CharSequence searchTerm) {
                         }
                         final String valueText = value.displayName.toLowerCase(locale);
 
-                        for (int k = 0; k < wordCount; k++) {
-                            if (valueText.contains(words[k])) {
+                        for (String word : words) {
```

Comment:
In this specific case I'd rather keep the existing code. Using an indexed loop avoids the creation of an `Iterator` instance.